docs/source/getting-started-client-mode.md

```diff
@@ -0,0 +1,98 @@
+  "display_name": "Spark 2.1 - Python (YARN Client Mode)",
+  "process_proxy": {
+    "class_name": "enterprise_gateway.services.processproxies.distributed.DistributedProcessProxy"
+  },
+  "env": {
+    "SPARK_HOME": "/usr/hdp/current/spark2-client",
+    "PYSPARK_PYTHON": "/opt/anaconda3/bin/python",
+    "PYTHONPATH": "${HOME}/.local/lib/python3.6/site-packages:/usr/hdp/current/spark2-client/python:/usr/hdp/current/spark2-client/python/lib/py4j-0.10.4-src.zip",
+    "SPARK_YARN_USER_ENV": "PYTHONUSERBASE=/home/yarn/.local,PYTHONPATH=${HOME}/.local/lib/python3.6/site-packages:/usr/hdp/current/spark2-client/python:/usr/hdp/current/spark2-client/python/lib/py4j-0.10.4-src.zip,PATH=/opt/anaconda2/bin:$PATH",
+    "SPARK_OPTS": "--master yarn --deploy-mode client --name ${KERNEL_ID:-ERROR__NO__KERNEL_ID} --conf spark.yarn.submit.waitAppCompletion=false",
+    "LAUNCH_OPTS": ""
+  },
+  "argv": [
+    "/usr/local/share/jupyter/kernels/spark_2.1_python_yarn_client/bin/run.sh",
+    "{connection_file}",
+    "--RemoteProcessProxy.response-address",
+    "{response_address}"
+  ]
+}
+```
+
+After making any necessary adjustments such as updating SPARK_HOME or other environment specific configuration, you now should have 
+a new Kernel available which will use Jupyter Enterprise Gateway to execute your notebook cell contents.
+
+**Configuring Kernels for Spark Standalone mode**
+
+The main difference between YARN Client and Standalone is the values used in SPARK_OPTS for the ```--master``` parameter.
+
+Please see below how a kernel.json would look like for integrating with Spark Standalone:
+ 
+ 
+```json
+{
+  "language": "python",
+  "display_name": "Spark 2.1 - Python (YARN Client Mode)",
+  "process_proxy": {
+    "class_name": "enterprise_gateway.services.processproxies.distributed.DistributedProcessProxy"
+  },
+  "env": {
+    "SPARK_HOME": "/usr/hdp/current/spark2-client",
+    "PYSPARK_PYTHON": "/opt/anaconda3/bin/python",
+    "PYTHONPATH": "${HOME}/.local/lib/python3.6/site-packages:/usr/hdp/current/spark2-client/python:/usr/hdp/current/spark2-client/python/lib/py4j-0.10.4-src.zip",
+    "SPARK_YARN_USER_ENV": "PYTHONUSERBASE=/home/yarn/.local,PYTHONPATH=${HOME}/.local/lib/python3.6/site-packages:/usr/hdp/current/spark2-client/python:/usr/hdp/current/spark2-client/python/lib/py4j-0.10.4-src.zip,PATH=/opt/anaconda2/bin:$PATH",
+    "SPARK_OPTS": "--master spark://127.0.0.1:7077  --name ${KERNEL_ID:-ERROR__NO__KERNEL_ID} --conf spark.yarn.submit.waitAppCompletion=false",
+    "LAUNCH_OPTS": ""
+  },
+  "argv": [
+    "/usr/local/share/jupyter/kernels/spark_2.1_python_yarn_client/bin/run.sh",
+    "{connection_file}",
+    "--RemoteProcessProxy.response-address",
+    "{response_address}"
+  ]
+}
+```
+
+After making any necessary adjustments such as updating SPARK_HOME or other environment specific configuration, you now should have 
+a new Kernel available which will use Jupyter Enterprise Gateway to execute your notebook cell contents.
```

Comment:
This needs to be mentioned somewhere in the above text.

Note: to gain complete round-robin functionality, the following steps must be taken:
1. Each of the kernelspec file-sets (`kernel.json` file and `bin` and `scripts` directories) must be copied to each of the nodes in the same folder as on the Enterprise Gateway server (e.g., `/usr/local/share/jupyter/kernels/<kernel-name>`).
2. The env variable `EG_REMOTE_HOSTS` must be set to a comma-separated set of host names indicating the hosts available for running kernels.  (This can be specified via the command line as well: `--EnterpriseGatewayApp.remote_hosts`)
3. The corresponding kernel runtime dependencies should be present on each of these nodes.